File: References/Java/SphrLibrary-Java/lib/src/main/java/sphrlibrary/entities/openmhealth/TimeFrame.java
package sphrlibrary.entities.openmhealth;

import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;

import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * 時間枠
 */
@Data
@NoArgsConstructor
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
public class TimeFrame {

    /**
     * 日付時刻
     */
    private String dateTime = null;

    /**
     * 時間間隔
     */
    private TimeInterval timeInterval = null;
}

File: SphrLibrary/Entities/OpenmHealth/TimeFrame.cs
using System.Runtime.Serialization;

namespace SphrLibrary.Entities.OpenmHealth
{
    /// <summary>
    /// Open mHealth TimeFrame リソースを表します。
    /// </summary>
    [DataContract()]
    public class TimeFrame
    {
        [DataMember(EmitDefaultValue = false)]
        public string? date_time = null; 

        [DataMember(EmitDefaultValue = false)]
        public TimeInterval? time_interval = null;

        public TimeFrame() { }
    }
}
